### PR TITLE
refactor: implement helper methods on `Property` types

### PR DESCRIPTION
**Pull request by vnepveu**
This pr simply creates some methods to make the code more clear when handling the custom types fields.

### CODE REVIEW

**Comment by vnepveu** [comment]
It is, I'm just waiting for it to be merged first

**Comment by adlerjohn** [comment]
Oh so you're going to merge `vnepveu/feat-enum-in-structs` into `master` first, then this branch into `master`?

**Comment by vnepveu** [comment]
> Oh so you're going to merge `vnepveu/feat-enum-in-structs` into `master` first, then this branch into `master`?

yes sir, unless you think it's bad practice?

**Comment by adlerjohn** [comment]
No it's fine, but to avoid merging by accident this should be a draft!

File: packages/fuels-types/src/lib.rs [hunk 1]
```diff
@@ -31,3 +32,27 @@ pub struct Property {
     pub type_field: String,
     pub components: Option<Vec<Property>>, // Used for custom types
 }
+
+// Both those constants are used to determine if a type field represents an `Enum` or a `Struct`.
+// Since it would have the format `struct foo` or `enum bar`, there is a whitespace.
+pub const STRUCT_KEYWORD: &str = "struct ";
+pub const ENUM_KEYWORD: &str = "enum ";
```
Comment (adlerjohn) [inline packages/fuels-types/src/lib.rs:39]: Do these need to be `pub`? Would `pub(crate)` suffice?

Comment (vnepveu) [reply packages/fuels-types/src/lib.rs:39]: you're right, some leftover from copy-paste

**Review by adlerjohn — COMMENTED** [review]
Looks fine to me, will defer to @digorithm

**Pull request merged**